WebAPIBookStore/WebAPIBookStore/Controllers/ProductController.cs

```diff
@@ -1,72 +1,46 @@
-﻿using AutoMapper;
-using Microsoft.AspNetCore.Mvc;
+﻿using Microsoft.AspNetCore.Mvc;
 using WebAPIBookStore.Dto;
-using WebAPIBookStore.Interfaces;
-using WebAPIBookStore.Models;
+using WebAPIBookStore.UseCase;
+
 namespace WebAPIBookStore.Controllers
 {
     [Route("api/Product")]
     [ApiController]
     public class ProductController : ControllerBase
     {
-        private readonly IProductRepository _productRepository;
-        private readonly ICategoryRepository _categoryRepository;
-        private readonly IMapper _mapper;
-        public ProductController(
-            IProductRepository productRepository,
-            ICategoryRepository categoryRepository,
-            IMapper mapper)
+        private readonly ProductUseCase _productUseCase;
+
+        public ProductController(ProductUseCase productUseCase)
         {
-            _productRepository = productRepository;
-            _categoryRepository = categoryRepository;
-            _mapper = mapper;
+            _productUseCase = productUseCase;
         }
 
         [HttpGet]
         public IActionResult GetProducts()
         {
-            var products = _productRepository.GetProducts();
-            if (products.Count <= 0)
-                return NotFound();
-
-            var productMaps = _mapper.Map<List<ProductDto>>(products);
-            return ModelState.IsValid ? Ok(productMaps) : BadRequest(ModelState);
+            var output = _productUseCase.Get();
+            return output.Error == "404" ? NotFound(output) : Ok(output);
```

Comment:
em define cái thằng 404 ra 1 file const để dùng chung các chỗ khác nhé.
Nên viết toán tử này theo kiểu case thành công trước -> case thất bại nhé

Reply:
lỗi tương tự các chỗ dưới